### PR TITLE
Feature/SKARE-16-DealerHand

### PR DESCRIPTION
**Pull request by erinyooey**
Implemented addCardsToDealer function feature where cards are added to the Dealer's Hand

### CODE REVIEW

File: CardGames.py
Comment (erinyooey) [inline CardGames.py:1]: This is the wrong version of CardGames.py.. Ignore this file.

**Review by sophianicolas — APPROVED** [review]
Logic:
- No major errors, logically sound.
- Keeping track of when the deck is empty and there are no more cards to give to the Dealer is helpful.

Test Coverage:
- The adding of test_unique() for ensuring a lack of duplicates is nice. This would be useful in keeping track of the cards used in a deck during each round.
- The tests are thorough and well written.

Style:
- No major style errors.
- The function and tests readable.

Functionality thoughts:
- The addCardsToDealer() among other functions can be extended to incorporate resetDeck() if players run out of cards, using multiple decks/dealers, or other poker strategies when it comes to ensuring used and reset cards are kept track of.

**Review by mccueki — APPROVED** [review]
Everything looks good so far. My only thought is that eventually we will need to keep track of player discard pile to be returned to the dealer. This can be done later.

**Comment by erinyooey** [comment]
> Logic:
> 
> * No major errors, logically sound.
> * Keeping track of when the deck is empty and there are no more cards to give to the Dealer is helpful.
> 
> Test Coverage:
> 
> * The adding of test_unique() for ensuring a lack of duplicates is nice. This would be useful in keeping track of the cards used in a deck during each round.
> * The tests are thorough and well written.
> 
> Style:
> 
> * No major style errors.
> * The function and tests readable.
> 
> Functionality thoughts:
> 
> * The addCardsToDealer() among other functions can be extended to incorporate resetDeck() if players run out of cards, using multiple decks/dealers, or other poker strategies when it comes to ensuring used and reset cards are kept track of.



> Everything looks good so far. My only thought is that eventually we will need to keep track of player discard pile to be returned to the dealer. This can be done later.



> Everything looks good so far. My only thought is that eventually we will need to keep track of player discard pile to be returned to the dealer. This can be done later.

**Comment by erinyooey** [comment]
> Logic:
> 
> * No major errors, logically sound.
> * Keeping track of when the deck is empty and there are no more cards to give to the Dealer is helpful.
> 
> Test Coverage:
> 
> * The adding of test_unique() for ensuring a lack of duplicates is nice. This would be useful in keeping track of the cards used in a deck during each round.
> * The tests are thorough and well written.
> 
> Style:
> 
> * No major style errors.
> * The function and tests readable.
> 
> Functionality thoughts:
> 
> * The addCardsToDealer() among other functions can be extended to incorporate resetDeck() if players run out of cards, using multiple decks/dealers, or other poker strategies when it comes to ensuring used and reset cards are kept track of.


Thank you for the insight! I will keep that in mind as I look into more edge cases and other functionality updates to make the game progress better as regards utilizing the resetDeck().

**Comment by erinyooey** [comment]
> Everything looks good so far. My only thought is that eventually we will need to keep track of player discard pile to be returned to the dealer. This can be done later.

Sounds good! That could definitely be something I could work on for my next task.

**Pull request merged**